src/main/java/com/example/demo/dto/PolygonDto.java

```diff
@@ -0,0 +1,63 @@
+package com.example.demo.dto;
+
+import com.example.demo.entity.Hole;
+import com.example.demo.entity.Point;
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import lombok.Value;
+
+import javax.validation.constraints.Size;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+@Value
+public class PolygonDto {
+    String name;
+    @Size(min = 3)
+    List<Point> points;
+    List<Hole> holes;
```

Comment:
There is a better way to implement holes.
have a look how it implemented in WKT and geojson